File: LeetCode_ProblemSet/Problem1669.c
#include <stdlib.h>

/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */


struct ListNode* mergeInBetween(struct ListNode* list1, int a, int b, struct ListNode* list2){
    struct ListNode *list2_head = list2, *list2_tail = list2;
    struct ListNode *pre = list1, *post = list1, *ans = list1;

    while (list2_tail->next != NULL) {
        list2_tail = list2_tail->next;
    }
    for (int i = 0; i < a - 1; i++) {
        pre = pre->next;
    }
    for (int i = 0; i < b + 1; i++) {
        post = post->next;
    }

    pre->next = list2_head;
    list2_tail->next = post;
    return ans;
}

File: LeetCode_ProblemSet/Problem1669.ts
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     val: number
 *     next: ListNode | null
 *     constructor(val?: number, next?: ListNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.next = (next===undefined ? null : next)
 *     }
 * }
 */

function mergeInBetween(list1: ListNode | null, a: number, b: number, list2: ListNode | null): ListNode | null {
    let list2_head: ListNode | null = list2, list2_tail: ListNode | null = list2;
    while (list2_tail.next !== null) {
        list2_tail = list2_tail.next;
    }

    let pre: ListNode | null = list1, post: ListNode | null = list1, ans: ListNode | null = list1;
    for (let i = 0; i < a - 1; i++) {
        pre = pre.next;
    }
    for (let i = 0; i < b + 1; i++) {
        post = post.next;
    }

    pre.next = list2_head;
    list2_tail.next = post;

    return ans;
};